### PR TITLE
feat(Theme): generateTheme customizations now supports body and title slots

### PR DESCRIPTION
**Pull request by lukelooker**
Addresses: https://b.corp.google.com/issues/183432420

## Developer Checklist [ℹ️](https://github.com/looker-open-source/components/blob/main//CONTRIBUTING.md#developer-checklist)

- [x] 🖼 Image Snapshot coverage
- [ ] 📚 Documentation updated
- [ ] 👾 Browsers tested
  - [x] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11

### CODE REVIEW

**Review by mdodgelooker — COMMENTED** [review]
Got a little confused by a comment. Also, would it be worth adding body & title color fields to the `ThemeEditor` in this PR, or were you thinking a follow-up?

File: packages/design-tokens/src/color/utils/generateColors.ts [hunk 1]
```diff
@@ -38,8 +38,24 @@ export const generateColors = (
 ): Colors => {
   const specifiable = { ...themeColors, ...pickBy(customColors, identity) }
 
+  /**
+   * If a theme customization only specifies a `text` color the `body` and
+   * `title` colors of the previous theme will be passed through.
+   *
+   * Instead, it should inferred that the new `text` color should be applied
+   * to the `body` & `title` slots unless those have been explicitly specified.
+   */
```
[the remark this answers is not in the document]
Comment (lukelooker) [reply packages/design-tokens/src/color/utils/generateColors.ts:47]: Took a stab at an improved wording:

```
  /**
   * If a theme customization only specifies a `text` color but does NOT
   * specify `body` or `title`  the previous/parent values will be erroneously
   * passed through.
   *
   * Instead, it should inferred that the new `text` color should be applied
   * to the `body` & `title` slots unless those have been explicitly specified
   * in `customColors`.
   */
```

**Comment by lukelooker** [comment]
Currently those values are specifiable within the "classic" theming experience as "text tile color" & "tile title color" - we added these slots to the theme to rationalize the use of those colors and currently do some funny things to inject the values from the classic theme editor (it works, but it's not the prettiest code). This change will now let values from the classic theme editor simply pass to `generateTheme` rather than having to modify the generated theme with these after the fact.

On the plus side the existing `text` color _does_ derive these so the colors "work" they're just not individually specifiable yet (`ThemeEditor`). As I start working on the ThemeEditor in earnest over the next few sprints we should be able to suss out what to expose (and not) as get everything nailed down.

TLDR - TBD

**Review by mdodgelooker — APPROVED** [review]
